.github/workflows/presubmit.yml

```diff
@@ -0,0 +1,876 @@
+name: Presubmit
+
+on: [push, pull_request]
+
+jobs:
+  format:
+    name: Code formatting
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          # repository: ${{ github.repository }} (default)
+          fetch-depth: 0
+      - name: Install clang-format
+        run: sudo apt-get install clang-format
+      - name: Check format
+        run: $GITHUB_WORKSPACE/scripts/check-format.sh
+          origin/`if [[ "${{ github.event_name }}" == "push" ]]; then echo "main"; else echo "${{ github.base_ref }}"; fi`
+          --binary clang-format
+
+  linux:
+    runs-on: ubuntu-latest
+    needs: format
+    defaults:
+      run:
+        shell: bash
+    strategy:
+      matrix:
+        CMAKE: [3.26.4]
+        COMPILER:
+          - C_NAME: gcc
+            CXX_NAME: g++
+            VER: 11
+          - C_NAME: gcc
+            CXX_NAME: g++
+            VER: 13
+          - C_NAME: clang
+            CXX_NAME: clang++
+            VER: 14
+          - C_NAME: clang
+            CXX_NAME: clang++
+            VER: 16
+        BIN: [64]
+        CXXSTD: [11, 17]
+        CONF:
+          - GEN: Unix Makefiles
+            CONFIG: Debug
+          - GEN: Unix Makefiles
+            CONFIG: Release
+          - GEN: Ninja Multi-Config
+            CONFIG: Release
+        IMAGE:
+          - streamhpc/opencl-sdk-base:ubuntu-22.04-20230717
```

Comment:
Checking: should this still be using a streamhpc image?

Reply:
Yes, we think that this is not a blocker for merge (the already merged Headers CI also uses these images). Is this correct?

The corresponding Dockerfiles are provided as a PR: https://github.com/KhronosGroup/DockerContainers/pull/32